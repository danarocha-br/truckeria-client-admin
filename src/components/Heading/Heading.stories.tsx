import { Story, Meta } from '@storybook/react/types-6-0';

import Heading from '.';

export default {
  title: 'Components/Heading',
  component: Heading,
  parameters: {
    layout: 'centered',
  },
} as Meta;

const Template: Story = (args) => <Heading {...args}>I am a title </Heading>;

export const Default = Template.bind({});
