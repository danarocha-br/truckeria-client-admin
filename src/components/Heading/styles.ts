import styled, { css } from 'styled-components';
import tw from 'twin.macro';

export const Heading = styled.h1`
  ${({ theme }) => css`
    ${tw`
      font-medium
      text-xl
    `}
    color: ${theme.text.default};
  `}
`;
