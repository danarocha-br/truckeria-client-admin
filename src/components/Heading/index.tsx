import * as S from './styles';

type HeadingProps = {
  children: React.ReactNode;
  className?: string;
};

const Heading = ({ children, className }: HeadingProps) => (
  <S.Heading className={className}>{children}</S.Heading>
);

export default Heading;
