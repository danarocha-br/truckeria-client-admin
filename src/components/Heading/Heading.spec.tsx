import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import Heading from '.';

describe('<Heading />', () => {
  it('should render the Heading', () => {
    const { container } = renderWithTheme(<Heading>Hey there</Heading>);

    expect(
      screen.getByRole('heading', { name: /Hey there/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
