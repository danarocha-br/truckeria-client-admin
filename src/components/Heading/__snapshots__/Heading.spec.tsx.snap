// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render the Heading 1`] = `
.c0 {
  font-weight: 500;
  font-size: 1.5rem;
  color: #20242A;
}

<h1
  class="c0"
>
  Hey there
</h1>
`;
